seisgo/noise.py

```diff
@@ -324,9 +316,9 @@ def smooth_source_spect(fft1,cc_method,sn):
         raise ValueError('no correction correlation method is selected at L59')
 
     return sfft1.reshape(N,Nfft2)
-#
-def do_correlation(sfile,win_len,step,maxlag,cc_method='xcorr',acorr_only=False,
-                    xcorr_only=False,substack=False,correct_orientation=False,substack_len=None,smoothspect_N=20,
+
+def do_correlation(sfile,win_len,step,maxlag,channel_pairs,cc_method='xcorr',acorr_only=False,
```

Comment:
similarly, make the usage compatible with the older version. give channel_pairs a default value. if not specified, it will correlate all pairs. you can do this by setting default as None and then check it in the function.

Reply:
do_correlation() is a wrapper function. I didn't change the input types at all. I just added some inputs which should be given by the user. So there should be no compatibility issue.
Once do_correlation is executed, both assemble_raw() and assemble_fft() will be executed automatically without any manual intervene. So there will be no compatibility issue as well unless someone wants to execute assemble_raw() or assemble_fft() separately.
For channel_pairs, I already organized the function as you described. You can refer to line 1059.

Reply:
what I meat was that if the user used the older without specifying channel_pairs, it should still run. So, to do that, you have to give it a default value.

Reply:
Yes, that what I mean. I already gave it a default value when defining the function. The default value is None.